pkg/server/api/bundle/v1/service.go

```diff
@@ -85,60 +81,42 @@ func (s *Service) AppendBundle(ctx context.Context, req *bundle.AppendBundleRequ
 	log := rpccontext.Logger(ctx)
 
 	if req.Bundle == nil {
-		log.Error("Invalid request: missing bundle")
-		return nil, status.Error(codes.InvalidArgument, "missing bundle")
+		return nil, api.MakeErr(log, codes.InvalidArgument, "missing bundle", nil)
 	}
 
 	log = log.WithField(telemetry.TrustDomainID, req.Bundle.TrustDomain)
 
 	td, err := spiffeid.TrustDomainFromString(req.Bundle.TrustDomain)
 	if err != nil {
-		log.WithError(err).Error("Invalid request: trust domain argument is not valid")
-		return nil, status.Errorf(codes.InvalidArgument, "trust domain argument is not valid: %v", err)
+		return nil, api.MakeErr(log, codes.InvalidArgument, "trust domain argument is not valid", err)
 	}
 
 	if s.td.Compare(td) != 0 {
-		log.Error("Invalid request: only the trust domain of the server can be appended")
-		return nil, status.Error(codes.InvalidArgument, "only the trust domain of the server can be appended")
-	}
-
-	dsResp, err := s.ds.FetchBundle(ctx, &datastore.FetchBundleRequest{
-		TrustDomainId: s.td.IDString(),
-	})
-	if err != nil {
-		log.WithError(err).Error("Failed to fetch server bundle")
-		return nil, status.Errorf(codes.Internal, "failed to fetch server bundle: %v", err)
-	}
-
-	if dsResp.Bundle == nil {
-		log.Error("Failed to fetch server bundle: not found")
-		return nil, status.Errorf(codes.NotFound, "failed to fetch server bundle: not found")
+		return nil, api.MakeErr(log, codes.InvalidArgument, "only the trust domain of the server can be appended", nil)
 	}
 
 	applyBundleMask(req.Bundle, req.InputMask)
 
 	dsBundle, err := api.ProtoToBundle(req.Bundle)
 	if err != nil {
-		log.WithError(err).Error("Invalid request: failed to convert bundle")
-		return nil, status.Errorf(codes.InvalidArgument, "failed to convert bundle: %v", err)
+		return nil, api.MakeErr(log, codes.InvalidArgument, "failed to convert bundle", err)
 	}
 
 	resp, err := s.ds.AppendBundle(ctx, &datastore.AppendBundleRequest{
 		Bundle: &common.Bundle{
 			TrustDomainId:  td.IDString(),
 			JwtSigningKeys: dsBundle.JwtSigningKeys,
 			RootCas:        dsBundle.RootCas,
+			RefreshHint:    dsBundle.RefreshHint,
```

Comment:
Append only operates on the JWT signing key and Root CAs.

I'm wondering if taking a `bundle` as input is even appropriate considering we'll ignore the sequence number and refresh hint...

Reply:
Good catch! I thought it was missing. Right, perhaps the `AppendBundleRequest` could take only the Root CAs and the JWT keys.